### PR TITLE
Avoid conflict between filters and side panel params

### PR DESCRIPTION
**Pull request by lorumic**
## Done

- Fixed a conflict between the filters query parameters and the side panel query parameters, that essentially broke the "Instance summary" side panel, making it impossible to open - see screen capture below:

Peek 2023-11-16 13-56

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @lorumic or @edlerd for access.
    - With a local copy of this branch, run as described [here](../HACKING.md#setting-up-for-development).
2. Perform the following QA steps:
    - Go to instance list
    - Click on a row in the table - check that it opens the side panel correctly
    - Try adding some filters - with the side panel still open
    - Try reloading the page
    - Check that both types of query parameters (those related to filters and those related to the side panel) are correctly preserved

### CODE REVIEW

File: src/pages/instances/InstanceSearchFilter.tsx [hunk 1]
```diff
@@ -44,50 +46,52 @@ const InstanceSearchFilter: FC<Props> = ({ instances }) => {
...
+
+    QUERY_PARAMS.forEach((param) => {
+      const values = searchValuesByLead(param);
+      if (values.length === 0) {
+        newParams.delete(param);
+      }
+      values.forEach((value) => {
+        if (!newParams.has(param, value)) {
+          newParams.append(param, value);
+        }
```
Comment (edlerd) [inline src/pages/instances/InstanceSearchFilter.tsx:66]: When having multiple chips for the same category selected and removing some (but not all), this doesn't remove the value from the params.

I.e. filter "running" or "stopped" and remove "stopped", this last change will not be reflected in the URL params.

**Review by edlerd — CHANGES_REQUESTED** [review]
Thanks for picking this up, I realized this earlier, good to see already a fix for it :)

Though one edge case is not working well yet, see comment below.

File: src/pages/instances/InstanceSearchFilter.tsx (hunk 1 above)
Comment (lorumic) [reply src/pages/instances/InstanceSearchFilter.tsx:63]: Good catch! Should be fixed now.

**Review by edlerd — APPROVED** [review]
LGTM

**Pull request merged**